src/app/feed/feed-card/feed-card.component.ts

```diff
@@ -126,4 +131,15 @@ export class FeedCardComponent implements OnInit {
 		}
 
 	}
+
+	private sanitizeandembedURLs(links) {
+		for( let i = 0; i < links.length; i++)
+		{
+			let videoid = links[i].match(/(?:https?:\/{2})?(?:w{3}\.)?youtu(?:be)?\.(?:com|be)(?:\/watch\?v=|\/)([^\s&]+)/);
+			if(videoid != null) {
+   				links[i] = "http://www.youtube.com/embed/" + videoid[1];
```

Comment:
Is there a way to just show the thumbnail of the video and then load the video in the modal, like twitter does, this can also solve the problem of the site loading slowly.